### PR TITLE
Fix for issue #599 (Rounding error on nanosecondDelta)

### PR DESCRIPTION
**Pull request by peter-int-veld**
Fixed by first executing the multiplication and then the addition.

### CODE REVIEW

File: src/data-type.js [hunk 1]
```diff
@@ -1280,13 +1280,14 @@ const TYPE = module.exports.TYPE = {
...
-        let ref, time = new Date(+parameter.value);
+        let time = new Date(+parameter.value);
         if (options.useUTC) {
           time = ((time.getUTCHours() * 60 + time.getUTCMinutes()) * 60 + time.getUTCSeconds()) * 1000 + time.getUTCMilliseconds();
         } else {
           time = ((time.getHours() * 60 + time.getMinutes()) * 60 + time.getSeconds()) * 1000 + time.getMilliseconds();
         }
-        time = (time / 1000 + ((ref = parameter.value.nanosecondDelta) != null ? ref : 0)) * Math.pow(10, parameter.scale);
+        time = Math.round(time * Math.pow(10, parameter.scale - 3));
+        time += Math.round((parameter.value.nanosecondDelta != null ? parameter.value.nanosecondDelta : 0) * Math.pow(10, parameter.scale));
```
Comment (tvrprasad) [inline src/data-type.js]: Please do the <code>Math.round()</code> after the addition. Like below:
```javascript
time = time * Math.pow(10, parameter.scale - 3);
time += (parameter.value.nanosecondDelta != null ? parameter.value.nanosecondDelta : 0) * Math.pow(10, parameter.scale);
time = Math.round(time);
```
Also, can you please add tests?

Thanks for fixing this!

**Comment by peter-int-veld** [comment]
Added a unit test for the fix.

**Comment by ErikJansenIRefact** [comment]
When will this fix be made available?

**Comment by ErikJansenIRefact** [comment]
Hi. 

We currently have another module: tedious-connectionpool which depends on tedious. We cannot deploy to production due to the issue with the nanoseconds. 

What should we do to get a review on this fix?
Thanks in advance for your suggestions.

**Pull request merged**